### PR TITLE
Bugfix scaffold order

### LINKED ISSUE #1044: Incorrect scaffolds being used to generate apps with flow setup
### Problem

flow version: Version: v1.0.2

The wrong scaffolds are being generated for several of the options available when using `flow setup` command.
Mobile and Unity scaffolds are mismatched.

📱 Mobile Scaffolds
   [5] Swift iOS simple example - iOS example demonstrating usage of FCL and Flow interactions.
   [6] Android simple example - Android example demonstrating usage of FCL and Flow interactions.

🏀 Unity Scaffolds
   [7] Simple Unity - Simple example demonstrating how to interact with the Flow network using Unity SDK.
   [8] Mobile Unity Game - Example words game built on Flow using the Unity SDK.

5, 6 Generate apps using **Simple Unity** and **Mobile Unity Game**
7, 8 Generate **Swift iOS simple example** and **Android simple example**

### Steps to Reproduce

```
$ brew install flow-cli
$ flow setup flow-app --scaffold
$ Enter the scaffold number:
```
Enter 5,6,7,8 and the wrong scaffold app is generated.

### Acceptance Criteria
Correct scaffolds are used to generate the apps.

### PR DESCRIPTION
**Pull request by sideninja**
Closes #1044 1044

## Description
Fix incorrect order on the scaffold listing. 

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: internal/super/setup.go [hunk 1]
```diff
@@ -91,15 +91,19 @@ func create(
 	pickedScaffold := scaffolds[0]
 
 	if setupFlags.Scaffold {
-		scaffoldList := make(map[string][]string)
-		for _, s := range scaffolds {
-			scaffoldList[s.Type] = append(
-				scaffoldList[s.Type],
-				fmt.Sprintf("%s - %s", output.Bold(s.Name), s.Description),
+		items := make([]util.ScaffoldItem, 0)
```
Comment (nialexsan) [inline internal/super/setup.go:94]: can we rename it back to `scaffoldList` or `scaffoldItems` to be more specific? `items` sounds very generic

File: internal/util/prompt.go [hunk 2]
```diff
@@ -678,7 +678,14 @@ func InstallPathPrompt(defaultPath string) string {
...
 
-func ScaffoldPrompt(logger output.Logger, availableScaffolds map[string][]string) int {
+type ScaffoldItem struct {
+	Index         int
+	Title         string
+	Category      string
+	assignedIndex int
+}
+
+func ScaffoldPrompt(logger output.Logger, items []ScaffoldItem) int {
```
Comment (nialexsan) [inline internal/util/prompt.go:688]: the same here with `items`

**Review by nialexsan — APPROVED** [review]
a few variable name suggestions

**Review by chasefleming — APPROVED** [review]

File: internal/util/prompt.go (hunk 2 above)
Comment (sideninja) [reply internal/util/prompt.go]: addressed

**Pull request merged**